blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -206,4 +268,28 @@ protected MetricData getHistogramsByGranularity(String tenantId,
 
         return data;
     }
+
+    private class UnitsGetter implements Callable {
+        private String tenantId;
+        private String metricName;
+
+        public UnitsGetter(String tenantId, String metricName) {
+            this.tenantId = tenantId;
+            this.metricName = metricName;
+        }
+
+        public String call() {
+            String unit = null;
+            try {
+                List<SearchResult> results = discoveryHandler.search(tenantId, metricName);
+                for (SearchResult res : results) {
```

Comment:
this whole bit is going to be incredibly slow! all these searches will kill the cluster! I'm not quite sold on using ES for the fetching of metadata, but if we were, this should be a simple `get`; it will be much faster.

will probably require a method added to the DiscoveryIO interface.